packages/core/src/Stores/gtm-store.js

```diff
@@ -96,14 +96,127 @@ export default class GTMStore extends BaseStore {
+     */
+    @action.bound
+    pushTransactionData(response, extra_data = {}) {
+        if (!this.is_gtm_applicable || this.root_store.client.is_virtual) return;
+        if (!response.transaction || !response.transaction.action) return;
+        if (!['deposit', 'withdrawal'].includes(response.transaction.action)) return;
+
+        const moment_now  = toMoment();
+        const storage_key = 'GTM_transactions';
+
+        // Remove values from prev days so localStorage doesn't grow to infinity
+        let gtm_transactions = JSON.parse(localStorage.getItem(storage_key)) || {};
+        if (Object.prototype.hasOwnProperty.call(gtm_transactions, 'timestamp')) {
+            if (moment_now.isAfter(epochToMoment(gtm_transactions.timestamp), 'day')) {
+                localStorage.removeItem(storage_key);
+                gtm_transactions = { timestamp: moment_now.unix() };
+            }
+        }
+        const transactions_arr = gtm_transactions.transactions || [];
+        if (!transactions_arr.includes(response.transaction.transaction_id)) {
+            const data = {
+                event           : 'transaction',
+                bom_account_type: this.root_store.client.account_type,
+                bom_today       : moment_now.unix(),
+                transaction     : {
+                    id     : response.transaction.transaction_id,
+                    type   : response.transaction.action,
+                    time   : response.transaction.transaction_time,
+                    amount : response.transaction.amount,
+                    balance: response.transaction.balance,
+                },
+            };
+            Object.assign(data, extra_data);
+            this.pushDataLayer(data);
+
+            transactions_arr.push(response.transaction.transaction_id);
+            gtm_transactions.transactions = transactions_arr;
+            gtm_transactions.timestamp    = gtm_transactions.timestamp || moment_now.unix();
+
+            localStorage.setItem(storage_key, JSON.stringify(gtm_transactions));
+        }
+    }
+
+    @action.bound
+    eventHandler(get_settings) {
+        if (!this.is_gtm_applicable) return;
+
+        const login_event       = localStorage.getItem('GTM_login');
+        const is_new_account    = localStorage.getItem('GTM_new_account') === '1';
+
+        localStorage.removeItem('GTM_login');
+        localStorage.removeItem('GTM_new_account');
+
+        const affiliate_token = Cookies.getJSON('affiliate_tracking');
+        if (affiliate_token) {
+            this.pushDataLayer({ affiliate_token: affiliate_token.t });
+        }
+
+        // Get current time (moment, set by server), else fallback to client time
+        const moment_now = toMoment();
+        const data = {
+            visitorId     : this.visitorId,
+            account_type  : this.root_store.client.account_type,
+            currency      : this.root_store.client.currency,
+            country       : get_settings.country,
```

Comment:
is this country based on IP or client's residence?

Reply:
@ashkanx It's based on client's residence.